src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -501,6 +501,28 @@ def common_encryption_checks(self, ds):
         if not ds.encrypted:
             raise CallError(f'{id} is not encrypted')
 
+    def get_dataset_by_path(self, path, datasets=None):
```

Comment:
I dont understand why we cant use libzfs.get_dataset_by_path ?

Reply:
It wouldn't work with say path `/mnt/test/abc` if the only dataset we have is `test`

Reply:
Are you sure?

truenas# mkdir /mnt/tank/foo/bar
truenas# python3
Python 3.8.3rc1 (default, Apr 30 2020, 07:33:30) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import libzfs
>>> libzfs.ZFS().get_dataset_by_path
<built-in method get_dataset_by_path of libzfs.ZFS object at 0x7fc8f1da3bd0>
>>> libzfs.ZFS().get_dataset_by_path('/mnt')
<libzfs.ZFSDataset name 'boot-pool/ROOT/13.0-MASTER-20200625-133242' type 'FILESYSTEM'>
>>> libzfs.ZFS().get_dataset_by_path('/mnt/tank')
<libzfs.ZFSDataset name 'tank' type 'FILESYSTEM'>
>>> libzfs.ZFS().get_dataset_by_path('/mnt/tank/foo')
<libzfs.ZFSDataset name 'tank/foo' type 'FILESYSTEM'>
>>> libzfs.ZFS().get_dataset_by_path('/mnt/tank/foo/bar')
<libzfs.ZFSDataset name 'tank/foo' type 'FILESYSTEM'>

Reply:
Interesting
```
>>> import libzfs;z=libzfs.ZFS()
>>> z.get_dataset_by_path('/mnt/test')
<libzfs.ZFSDataset name 'test' type 'FILESYSTEM'>
>>> z.get_dataset_by_path('/mnt/test/abc')
cannot open '/mnt/test/abc': No such file or directory
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "libzfs.pyx", line 1199, in libzfs.ZFS.get_dataset_by_path
libzfs.ZFSException: Dataset with path /mnt/test/abc not found
>>>
```
I'll look closely :)

Reply:
Oh, I think I found the issue :) My bad, thank you for pointing it out